### PR TITLE
fix(vrl): Fix parse_xml handling of single node children

### PR DESCRIPTION
**Pull request by jszwedko**
Previously nodes with a single child element did not have attributes
associated.

Fixes: #11901

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: lib/vrl/stdlib/src/parse_xml.rs [hunk 1]
```diff
@@ -300,12 +300,21 @@ fn process_node<'a>(node: Node, config: &ParseXmlConfig<'a>) -> Value {
...
         for n in node.children().into_iter().filter(|n| !n.is_comment()) {
-            // Use the default tag name if blank.
-            let name = if n.tag_name().name() == "" {
-                config.text_key.to_string()
-            } else {
-                n.tag_name().name().to_string()
+            let name = match n.node_type() {
+                NodeType::Element => n.tag_name().name().to_string(),
+                NodeType::Text => config.text_key.to_string(),
+                _ => unreachable!("shouldn't be other XML nodes"),
```
Comment (jszwedko) [inline lib/vrl/stdlib/src/parse_xml.rs:317]: This change is unrelated, I just found the existing code hard to reason about. I'm not crazy about the `unreachable` here but we were already doing that below. This function should only be called with elements or text nodes.

**Review by pablosichert — APPROVED** [review]
Looks good 👍

I'd add the provided test case from the issue as well:

```
<root><node attr="value"><message>bar</message></node><node attr="value"><message>baz</message></node></root>
```
```
{"root":{"node":[{"@attr":"value","message":"bar"},{"@attr":"value","message":"baz"}]}}
```

**Comment by jszwedko** [comment]
> Looks good 👍
> 
> I'd add the provided test case from the issue as well:
> 
> ```
> <root><node attr="value"><message>bar</message></node><node attr="value"><message>baz</message></node></root>
> ```
> 
> ```
> {"root":{"node":[{"@attr":"value","message":"bar"},{"@attr":"value","message":"baz"}]}}
> ```

That was already working correctly, but yes, I do see that none of the tests specifically cover that. I'll add one.
